tensorboard/data/server/server.rs

```diff
@@ -386,8 +387,32 @@ mod tests {
         );
     }
 
-    // TODO(@bileschi): Add a test for the case of multiple different types of
-    // tags once sample CommitBuilders exist for plugins beyond "scalar".
+    #[tokio::test]
+    async fn test_list_plugins_multiple_timeseries_different_types() {
+        let mut custom_metadata = pb::SummaryMetadata::default();
+        let mut plugin_data = pb::summary_metadata::PluginData::default();
+        plugin_data.plugin_name = "custom_scalars".to_string();
+        custom_metadata.plugin_data = Some(plugin_data);
+        let commit = CommitBuilder::new()
+            .scalars("train", "xent2", |b| b.build())
+            .scalars("train", "xent", |mut b| {
+                b.metadata(Some(Box::new(custom_metadata))).build()
+            })
+            .build();
+        let handler = sample_handler(commit);
+        let req = Request::new(data::ListPluginsRequest {
+            experiment_id: "123".to_string(),
+        });
+        let res = handler.list_plugins(req).await.unwrap().into_inner();
+        assert_eq!(
+            res.plugins
+                .into_iter()
+                .map(|p| p.name)
+                .collect::<Vec<_>>()
+                .sort(),
+            vec!["custom_scalars", "scalars"].sort()
+        );
```

Comment:
`sort()` returns unit, not the vector itself (same as Python):

```rust
fn main() {
    assert_eq!(vec![1, 2].sort(), vec![3, 4].sort()); // passes!
}
```

```python
assert [1, 2].sort() == [3, 4].sort()  # passes!
```

([Clippy catches this][unit-cmp], but it looks like we don’t run Clippy
on test code. I’ll fix that separately, since we have a few existing
Clippy warnings in tests.)

[unit-cmp]: https://rust-lang.github.io/rust-clippy/master/index.html#unit_cmp

You’ll want to sort the vectors and then compare them (as variables).

Reply:
updated to capture the sorted lists